src/static_handler.rs

```diff
@@ -49,22 +48,18 @@ impl Handler for Static {
             //
             // rust-url automatically turns an empty string in the last
             // slot in the path into a trailing slash.
-            redirect_path.path.push("".into_string());
+            redirect_path.path.push("".to_string());
 
-            return Ok(Response::new().set(Status(status::MovedPermanently))
-                          .set(Body(format!("Redirecting to {}", redirect_path)))
-                          .set(Redirect(redirect_path)));
+            return Ok(Response::with((status::MovedPermanently,
+                                      format!("Redirecting to {}", redirect_path),
+                                      Redirect(redirect_path))));
         }
 
         match requested_path.get_file() {
-            Some(path) =>
-                Ok(Response::new()
-                       .set(Status(status::Ok))
-                       // Won't panic because we know the file exists from get_file
-                       .set(Body(path))),
-            None =>
-                // If no file is found, return a 404 response.
-                Ok(Response::new().set(Status(status::NotFound)).set(Body("File not found")))
+            // Won't panic because we know the file exists from get_file
+            Some(path) => Ok(Response::with((status::Ok, path))),
+            // If no file is found, return a 404 response.
+            None => Ok(Response::with((status::NotFound, "File not found")))
```

Comment:
I wouldn't bother with `FileStatError` for now. It does seem reasonable to have at some point though, since the type is the only thing receivers have to go off.